### PR TITLE
change column type of archived background data from json to text

### PR DESCRIPTION
**Pull request by boris-dimitrov**
json is new in MySQL 5.7.8 and thus not yet supported on Aurora

### CODE REVIEW

File: db/schema.rb [hunk 1]
```diff
@@ -156,6 +156,16 @@
     t.index ["user_id"], name: "index_projects_users_on_user_id"
   end
 
+  create_table "reports", force: :cascade, options: "ENGINE=InnoDB DEFAULT CHARSET=utf8 COLLATE=utf8_unicode_ci" do |t|
```
Comment (boris-dimitrov) [inline db/schema.rb]: Wow what happened here...  I didn't expect to get this table back, sorry!    I think it was left over in my local DB.

Have to redo this.

Comment (boris-dimitrov) [reply db/schema.rb]: I think what happened here is that the old tables still exist on alpha.   So alpha doesn't have the latest code or somehow hasn't run the migrations that delete those tables.

**Comment by cdebourcy** [comment]
I think you also need to change from "as_json" to "to_json" in line 44 of background_controller so that the object is actually a string, not json

File: app/controllers/backgrounds_controller.rb [hunk 2]
```diff
@@ -41,7 +41,7 @@ def create
   # PATCH/PUT /backgrounds/1
   # PATCH/PUT /backgrounds/1.json
   def update
-    current_data = @background.as_json(include: [{ pipeline_outputs: { only: [:id, :pipeline_run_id] } },
+    current_data = @background.to_json(include: [{ pipeline_outputs: { only: [:id, :pipeline_run_id] } },
```
Comment (boris-dimitrov) [inline app/controllers/backgrounds_controller.rb]: ok, changed that

how do i test it?   does it run on every background modification?

Comment (cdebourcy) [reply app/controllers/backgrounds_controller.rb]: Yes - you can test it by modifying any background (go to localhost:3000/backgrounds/:id/edit) and checking that an ArchivedBackground record gets created correctly

Comment (boris-dimitrov) [reply app/controllers/backgrounds_controller.rb]: thanks, confirmed it works

**Pull request merged**